src/mscorlib/shared/System/Memory.cs

```diff
@@ -257,10 +256,9 @@ public bool TryGetArray(out ArraySegment<T> arraySegment)
                     return true;
                 }
             }
-            else
+            else if (_object is T[] arr)
             {
-                T[] arr = _object as T[];
-                if (typeof(T) != typeof(char) || arr != null)
+                if (typeof(T) != typeof(char))
```

Comment:
This check should be deleted.

Reply:
I am curious, why did we have the check to begin with? 
`_object as T[]` => can this end up casting a string to char[]?
`_object is T[] arr` => will this be false for string object?

Reply:
To avoid the cost of the check when T wasn't char (it would be compiled away by the JIT).